src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -3812,6 +3854,110 @@ bool emberAfDoorLockClusterClearHolidayScheduleCallback(
+    if (!SupportsAliroProvisioning(endpointID))
+    {
+        ChipLogProgress(Zcl, "[SetAliroReaderConfig] Aliro Provisioning is not supported [endpointId=%d]", endpointID);
+        commandObj->AddStatus(commandPath, Status::UnsupportedCommand);
+        return;
+    }
+
+    Delegate * delegate = GetDelegate(endpointID);
+    VerifyOrReturn(delegate != nullptr, ChipLogError(Zcl, "Delegate is nullptr"));
+
+    uint8_t buffer[kAliroReaderVerificationKeyMaxSize];
+    MutableByteSpan aliroReaderVerificationKey(buffer);
+
+    // If Aliro reader verification key attribute is not null, we can't set a new reader config without clearing the previous one,
+    // return INVALID_IN_STATE.
+    if (CHIP_NO_ERROR == delegate->GetAliroReaderVerificationKey(aliroReaderVerificationKey) &&
+        aliroReaderVerificationKey.data() != nullptr)
+    {
+        ChipLogProgress(Zcl, "[SetAliroReaderConfig] Aliro reader verification key is not null. Return INVALID_IN_STATE");
+        commandObj->AddStatus(commandPath, Status::InvalidInState);
+        return;
+    }
+
+    // If Aliro BLE UWB feature is supported and groupResolvingKey is not provided in the command, return INVALID_COMMAND.
+    if (SupportsAliroBLEUWB(endpointID) && !groupResolvingKey.HasValue())
+    {
+        ChipLogProgress(Zcl, "[SetAliroReaderConfig] Aliro BLE UWB supported but Group Resolving Key is not provided");
+        commandObj->AddStatus(commandPath, Status::InvalidCommand);
+        return;
+    }
+
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    if (!emberAfPluginDoorLockSetAliroReaderConfig(endpointID, signingKey, verificationKey, groupIdentifier, groupResolvingKey))
+    {
+        ChipLogProgress(Zcl, "[SetAliroReaderConfig] Unable to set aliro reader config [endpointId=%d]", endpointID);
+        status = EMBER_ZCL_STATUS_FAILURE;
+    }
+    sendClusterResponse(commandObj, commandPath, status);
+}
+
+void DoorLockServer::clearAliroReaderConfigCommandHandler(CommandHandler * commandObj, const ConcreteCommandPath & commandPath)
+{
+    EndpointId endpointID = commandPath.mEndpointId;
+    ChipLogProgress(Zcl, "[ClearAliroReaderConfig] Incoming command [endpointId=%d]", endpointID);
+
+    // If Aliro Provisioning feature is not supported, return UNSUPPORTED_COMMAND.
+    if (!SupportsAliroProvisioning(endpointID))
+    {
+        ChipLogProgress(Zcl, "[ClearAliroReaderConfig] Aliro Provisioning is not supported [endpointId=%d]", endpointID);
+        commandObj->AddStatus(commandPath, Status::UnsupportedCommand);
+        return;
+    }
+
+    Delegate * delegate = GetDelegate(endpointID);
+    VerifyOrReturn(delegate != nullptr, ChipLogError(Zcl, "Delegate is nullptr"));
+
+    uint8_t buffer[kAliroReaderVerificationKeyMaxSize];
+    MutableByteSpan aliroReaderVerificationKey(buffer);
+
+    // If Aliro reader verification key attribute is null, there is no reader config to clear. Return INVALID_IN_STATE.
```

Comment:
Wait, why?  There's nothing in the spec that says to do this.

Reply:
I thought I was making up something. should have double checked the spec. This is on me.